### PR TITLE
Describe runtime agent contract

### PR DESCRIPTION
**Pull request by crabtree**
**Description**

Changes proposed in this pull request:

- Describe runtime agent contract

**Related issue(s)**

See also #47

### CODE REVIEW

File: components/director/pkg/graphql/schema.graphql [hunk 1]
```diff
@@ -318,6 +318,13 @@ type HealthCheck {
     timestamp: Timestamp!
 }
 
+# RuntimeConfiguration
+
+type RuntimeConfiguration {
```
Comment (PK85) [inline components/director/pkg/graphql/schema.graphql]: Consider Moving to Runtime domain. We will have less models, queries.

File: docs/architecture/runtime-agent-contract.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+## Renewing trusted connection
+
+Depending on the type of trusted connection, during the runtime lifecycle, it may be required to periodically [renew trusted connection](./establishing-trusted-connection.md#client-certificate-flow---certificate-renewal).
+
+## Configuring the runtime
+
+Runtime Agent periodically requests for the configuration of its Runtime from the Management Plane. Changes in the configuration for the Runtime are applied by the Runtime Agent on the Runtime. To fetch the Runtime configuration the Runtime Agent calls `runtimeConfiguration(id: ID!)` query offered by [Director](../terminology.md#mp-director). The response for the query contains list of Applications assigned for the Runtime.
+
+Runtime Agent reports the Runtime eventing configuration to the Director. Runtime entity stores URL of the runtime. Based on the configuration the Eventing URL and other useful URLs, like console UI URL, can be computed.
```
Comment (PK85) [inline docs/architecture/runtime-agent-contract.md]: Krystian mentioned additional requirement, that maybe we coudl handle by labels. 
- Namely, not only URL from events, but we ned to expose runtime console UI and other URLs, maybe even kyma/runtime versions etc.
- We can realize that already by LabelDefintions with schemas. We need to decide that maybe those LD should be global, not removable by tenants.
- All LD values are computed onky by runtime Agent and passed to MP
- We can even consider adding new LD by runtime Agent (if new ones supported by runtime)

Comment (crabtree) [reply docs/architecture/runtime-agent-contract.md]: This should be covered in #106

File: docs/architecture/runtime-agent-contract.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# Runtime Agent contract
+
+## Overview
+
+This document describes contract between [Management Plane](../terminology.md#management-plane) and [Runtime Agent](../terminology.md#runtime-agent). Main responsibilities of the Runtime Agent are:
+- [Establishing trusted connection](#establishing-trusted-connection)
+- [Renewing trusted connection](#renewing-trusted-connection)
+- [Configuring applications](#configuring-applications)
```
Comment (franpog859) [inline docs/architecture/runtime-agent-contract.md]: The link should look like this to work correctly:
`- [Configuring the runtime](#configuring-the-runtime)`

**Comment by crabtree** [comment]
> Shouldn't the health checks mentioned [here](https://github.com/kyma-incubator/compass/blob/0c12b0f53f7e113f97e4954000d3e84beadc01a6/docs/terminology.md#runtime-agent) be involved in this contract?

As agreed with @PK85 the health checks will be defined in the next iteration of Runtime Agent Contract. For now, we do not focus on this area.

File: docs/architecture/runtime-agent-contract.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+## Overview
+
+This document describes contract between [Management Plane](../terminology.md#management-plane) and [Runtime Agent](../terminology.md#runtime-agent). Main responsibilities of the Runtime Agent are:
+- [Establishing trusted connection](#establishing-trusted-connection)
+- [Renewing trusted connection](#renewing-trusted-connection)
+- [Configuring the runtime](#configuring-the-runtime)
+
+## Establishing trusted connection
+
+Runtime Agent automates the process of [establishing trusted connection](./establishing-trusted-connection.md#establishing-trusted-connection) between [Runtime](../terminology.md#runtime) and Managemt Plane. To initialy trigger the process the [Runtime Provisioner](../terminology.md#mp-runtime-provisioner) creates the instance of CustomResource that holds the one-time token to enable the pairing. Runtime Agent triggered by the creation of CustomResource performs the pairing process and in the end the trusted connection between the Runtime and the Management Plane is established. At this point further communication between Runtime Agent and Management Plane, and [Application](../terminology.md#application) and Runtime is possible.
```
Comment (PK85) [inline docs/architecture/runtime-agent-contract.md]: That description explains the flow and then implementation details for Kyma as a Runtime. That's good. But we need also genberal contract. That means what information will be included in the token etc. 
For example, we will need information about tenant, scenario etc.

File: docs/architecture/runtime-agent-contract.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+## Renewing trusted connection
+
+Depending on the type of trusted connection, during the runtime lifecycle, it may be required to periodically [renew trusted connection](./establishing-trusted-connection.md#client-certificate-flow---certificate-renewal).
+
+## Configuring the runtime
+
+Runtime Agent periodically requests for the configuration of its Runtime from the Management Plane. Changes in the configuration for the Runtime are applied by the Runtime Agent on the Runtime. To fetch the Runtime configuration the Runtime Agent calls `applicationsForRuntime(runtimeId: ID!)` query offered by [Director](../terminology.md#mp-director). The response for the query contains list of Applications assigned for the Runtime.
```
Comment (PK85) [inline docs/architecture/runtime-agent-contract.md]: Add a first sentence that Runtime will get condigurations and update itself in management plane with details about a runtime.

File: docs/architecture/runtime-agent-contract.md (hunk 4 above)
Comment (crabtree) [reply docs/architecture/runtime-agent-contract.md]: @PK85 Why do we need information like tenant and labeling (eg.scenario) on the runtime agent? I assume that runtime agent need only the Runtime ID. Using runtime ID agent fetches the runtime configuration from Management Plane.

**Pull request merged**